src/databricks/labs/ucx/hive_metastore/mapping.py

```diff
@@ -103,14 +125,90 @@ def skip_table(self, backend: StatementExecutionBackend, schema: str, table: str
             if "[SCHEMA_NOT_FOUND]" in str(nf):
                 logger.error(f"Failed to apply skip marker for Schema {schema}. Schema not found.")
             else:
                 logger.error(nf)
         except BadRequest as br:
             logger.error(br)
+
+    def _get_upgraded_tables(self) -> dict[str, TableInfo]:
+        upgraded_tables = {}
+        for catalog in self._ws.catalogs.list():
+            if not catalog.name:
+                continue
+            for schema in self._ws.schemas.list(catalog_name=catalog.name):
+                if not schema.name:
+                    continue
+                for table in self._ws.tables.list(catalog_name=catalog.name, schema_name=schema.name):
+                    if table.properties is not None and "upgraded_from" in table.properties:
+                        upgraded_tables[table.properties["upgraded_from"].lower()] = table
+        return upgraded_tables
+
+    def _get_databases_in_scope(self, databases: set[str]):
+        tasks = []
+        for database in databases:
+            tasks.append(partial(self._get_database_in_scope_task, database))
+        return Threads.strict("checking databases for skip property", tasks)
+
+    def _get_database_in_scope_task(self, database: str) -> str | None:
+        describe = {}
+        for value in self._backend.fetch(f"DESCRIBE SCHEMA EXTENDED {database}"):
+            describe[value["database_description_item"]] = value["database_description_value"]
+        if self.UCX_SKIP_PROPERTY in TablesCrawler.parse_database_props(describe.get("Properties", "").lower()):
+            logger.info(f"Database {database} is marked to be skipped")
+            return None
+        return database
+
+    def _get_tables_in_scope(self, rules: list[Rule], databases_in_scope: set[str], crawled_tables: list[Table]):
+        crawled_tables_keys = {crawled_table.key: crawled_table for crawled_table in crawled_tables}
+        upgraded_tables = self._get_upgraded_tables()
+        tables_to_check = []
+        for rule in rules:
+            if rule.as_hms_table_key not in crawled_tables_keys:
+                logger.info(f"Table {rule.as_hms_table_key} in the mapping doesn't show up in assessment")
+                continue
+            if rule.as_hms_table_key in upgraded_tables:
+                logger.info(f"Table {rule.as_hms_table_key} was migrated to {rule.as_uc_table_key} and will be skipped")
+                continue
+            if rule.src_schema not in databases_in_scope:
+                logger.info(f"Table {rule.as_hms_table_key} is in a database that was marked to be skipped")
+                continue
+            tables_to_check.append(TableToMigrate(crawled_tables_keys[rule.as_hms_table_key], rule))
+        tasks = []
+        for table_to_check in tables_to_check:
+            tasks.append(partial(self._get_table_in_scope_task, table_to_check))
+        return Threads.strict("checking all database properties", tasks)
```

Comment:
```suggestion
            tables_to_check.append(partial(self._get_table_in_scope_task, 
                TableToMigrate(crawled_tables_keys[rule.as_hms_table_key], rule)))
        return Threads.strict("getting tables in scope", tables_to_check)
```
you don't need an extra loop here. also the name of the thread pool was incorrect

Reply:
Removed